hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ReconTaskControllerImpl.java

```diff
@@ -741,16 +744,16 @@ private void processReInitializationEvent(ReconTaskReInitializationEvent event)
           resetRetryCounters();
           LOG.info("Completed async task reinitialization");
         }
+        
+        // Clean up checkpoint files after use (database connections are automatically closed by try-with-resources)
+        cleanupCheckpointFiles(checkpointedOMMetadataManager);
```

Comment:
It's good to separate the file cleanup and the closing of `ReconOmMetadataManagerImpl`, but the file cleanup itself should still be put in a finally block to ensure it runs properly.